src/gui/AutomationPatternView.cpp

```diff
@@ -218,6 +219,10 @@ void AutomationPatternView::constructContextMenu( QMenu * _cm )
 	}
 
 	_cm->addSeparator();
+	_cm->addAction( embed::getIconPixmap( "colorize" ),
+			tr( "Set clip color" ), this, SLOT( changeClipColor() ) );
+	_cm->addAction( embed::getIconPixmap( "colorize" ),
+			tr( "Use track color" ), this, SLOT( useTrackColor() ) );
```

Comment:
These actions are repeated in each different clip view. It might be a good idea to move them to `TrackContentObjectView::constructContextMenu`, and then call the base `TrackContentObjectView::constructContextMenu` from each overridden version.

Reply:
I've migrated these to `TCOView`'s defintion, however I had to leave it in `SampleTCO` because it has its own construct menu definition (which overrides `TCOView`'s).